notification/src/handlers/amqpHandler.ts

```diff
@@ -0,0 +1,176 @@
+import { StorageHandler } from "./storageHandler"
+import { connect, Connection, ConsumeMessage } from "amqplib/callback_api"
+import { TransformationEvent } from '../interfaces/transformationResults/transformationEvent';
+import JSNotificationService from '../jsNotificationService';
+import VM2SandboxExecutor from "../vm2SandboxExecutor";
+import { CONFIG_TYPE } from "../models/notificationConfig";
+
+/**
+ * This class handles the communication with the AMQP service (rabbitmq)
+ * It interacts with this channels:
+ *
+ *      * Notification Channel:
+ *       ----------------------
+ *       A channel to notify the notification service that a transformation is done.
+ *       (see TransformationEvent for details of the Event).
+ *      
+ */
+export class AmqpHandler{
+    notifQueueName = process.env.AMQP_NOTIFICATION_QUEUE!     // Queue name of the Job Queue
+
+    storageHandler : StorageHandler
+    executor : VM2SandboxExecutor
+
+    /**
+     * Default constructor.
+     * 
+     * @param storageHandler    StorageHandler to get corresponding notification configs
+     * @param executor          Sandboxexecutor to run condition evaluations
+     */
+    constructor(storageHandler : StorageHandler, executor : VM2SandboxExecutor) {
+        this.storageHandler = storageHandler
+        this.executor = executor
+    }
+    /**
+     * Connects to Amqp Service and initializes a channel
+     * 
+     * @param retries   Number of retries to connect to the notification-config db
+     * @param backoff   Time to wait until the next retry
+     */
+    public async connect(retries: number, backoff: number) {
+        const rabbit_url = process.env.AMQP_SERVICE_HOST;
+        const rabbit_usr = process.env.AMQP_SERVICE_USER;
+        const rabbit_password = process.env.AMQP_SERVICE_PWD;
+        const rabit_amqp_url = 'amqp://' + rabbit_usr + ':' + rabbit_password + '@' + rabbit_url;
+
+        console.log("URL: " + rabit_amqp_url)
+
+        var established: boolean = false    // amqp service connection result
+        const handler: AmqpHandler = this   // for ability to access methods and members in callback
+        let errMsg: string = ''             // Error Message to be shown after final retry
+        
+        for (let i = 1; i <= retries; i++) {
+            await this.backOff(backoff)
```

Comment:
I think we should move this to the end of the method. Otherwise, we always wait, even when there is no error.